File: c++/mul.cc
#include "mul.h"
#include <iostream>

using namespace std;

void mul::ejecutar(stack<int>& pila){
    int valor1 = pila.top();
    pila.pop();
    int valor2 = pila.top(); 
    pila.pop();
    pila.push(valor1 * valor2);
}

int mul::modificar(int pc){
    return pc + 1;
}

std::string mul::mostrar() {
    return " mul" ;
}

File: java/mul.java
import java.util.Stack;

class mul extends Instruccion{

    @Override
    public void ejecutar(Stack<Integer> pila){
        int valor1 = pila.pop(); //desapilo el primer operando
        int valor2 = pila.pop(); //desapilo el segundo operando
        pila.push(valor1 * valor2); //apilo el resultado de la multiplicación de los dos valores que estaban en la cima
    }

    @Override
    public int incrementarPc(int pc){
        return pc + 1; //como en las anteriores
    }

    @Override
    public String mostrar(){
        return "    mul";
    }
}
